### PR TITLE
OSD-24723 unit testcases for code under /pkg/ocm/

### PR DESCRIPTION
**Pull request by vaidehi411**
### What type of PR is this?
Unit Testcases for code under /pkg/ocm/

### What this PR does / why we need it?
More Code Coverage

### Which Jira/Github issue(s) this PR fixes?
[OSD-24723](https://issues.redhat.com//browse/OSD-24723)
_Fixes #_

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

**Comment by shitaljante** [comment]
Changes looks good to me. 
There is one method `PostServiceLog` in ocm/client.go which is added as part of PHC epic. We can try to cover that as well.

**Comment by shitaljante** [comment]
> Changes looks good to me. There is one method `PostServiceLog` in ocm/client.go which is added as part of PHC epic. We can try to cover that as well.

This was added recently under "Upgrade Experience Enhancements" epic we will cover unit test for this separately.

/lgtm

File: pkg/scaler/scaling_test.go [hunk 1]
```diff
@@ -635,5 +749,86 @@ var _ = Describe("Node scaling tests", func() {
...
+
+			gomock.InOrder(
+				mockDrainStrategy.EXPECT().Execute(gomock.Any(), gomock.Any()).Return(strategy, nil).AnyTimes(),
+				mockDrainStrategy.EXPECT().HasFailed(gomock.Any(), gomock.Any()).Return(true, nil),
+			)
+			result, err := handleDrainStrategy(mockKubeClient, mockDrainStrategy, *nodes, logger)
+			Expect(err).Error()
+			Expect(result).To(BeFalse())
+		})
+		It("Handle Drain strategy has failed", func() {
```
Comment (shitaljante) [inline pkg/scaler/scaling_test.go:805]: Unit case name is duplicate here, we can add more specific test name ex. Handle Drain strategy has failed due to timeout

**Review by shitaljante — CHANGES_REQUESTED** [review]

File: pkg/scaler/scaling_test.go (hunk 1 above)
Comment (vaidehi411) [reply pkg/scaler/scaling_test.go]: Updated the testcase name.

**Comment by ravitri** [comment]
/approve

Thanks a lot Vaidehi for the PR and Shital and Tony for review :+1:

**Comment by ravitri** [comment]
/approve
/lgtm

Thanks a lot Vaidehi for the PR and Shital and Tony for review 👍

**Pull request merged**